src/du_parameters/frequency.py

```diff
@@ -0,0 +1,526 @@
+        return Frequency(super().__sub__(Decimal(other)))
+
+    def __rsub__(self, other: "Frequency | Decimal | str | int") -> "Frequency":
+        """Subtract Frequency from others.
+
+        Args:
+            other (Frequency | Decimal | str | int): The value to reversely subtract from.
+
+        Returns:
+            Frequency: The resulting Frequency after the subtraction.
+        """
+        if not isinstance(other, (Frequency, Decimal, int, str)):
+            raise NotImplementedError(f"Unsupported type for subtraction: {type(other)}")
+        return Frequency(super().__sub__(Decimal(other)))
+
+    def __mul__(self, other: "Frequency | Decimal | str| int") -> "Frequency":
+        """Multiply Frequency by others.
+
+        Args:
+            other (Frequency | Decimal | str | int): The value to multiply by.
+
+        Returns:
+            Frequency: The resulting Frequency after the multiplication.
+        """
+        if not isinstance(other, (Frequency, Decimal, int, str)):
+            raise NotImplementedError(f"Unsupported type for multiplication: {type(other)}")
+        return Frequency(super().__mul__(Decimal(other)))
+
+    def __truediv__(self, other: "Decimal | str | int") -> "Frequency":
+        """Divide Frequency by others.
+
+        Args:
+            other (Frequency | Decimal | str | int): The value to divide by.
+
+        Returns:
+            Frequency: The resulting Frequency after the division.
+        """
+        if not isinstance(other, (Decimal, int, str)):
+            raise NotImplementedError(f"Unsupported type for division: {type(other)}")
+        if other == 0:
+            raise ZeroDivisionError("Division by zero is not allowed.")
+        return Frequency(super().__truediv__(Decimal(other)))
+
+    def __repr__(self) -> str:
+        """Return the Frequency as a string for development purposes."""
+        return f"Frequency({self.to_eng_string()})"
+
+    def __str__(self) -> str:
+        """Return the Frequency as a string."""
+        return self.to_eng_string()
+
+
+class ARFCN:
+    """Represents an Absolute Radio Frequency Channel Number (ARFCN) used in 5G.
+
+    Provides arithmetic operations and conversions from frequencies.
+
+    Args:
+        channel (ARFCN | int): The input ARFCN instance.
+
+    Raises:
+        TypeError: If the ARFCN is not an integer or ARFCN instance.
+        ValueError: If the ARFCN is not within the valid range for ARFCN.,
+        NotImplementedError: If the other value is not an ARFCN, int or Decimal.
+    """
+
+    def __init__(self, channel: "int | ARFCN | Decimal"):
+        if not isinstance(channel, (int, ARFCN | Decimal)):
+            raise TypeError("Channel must be an integer or ARFCN instance.")
+        if channel < 0 or channel > 3279165:  # type: ignore[operator]
+            raise ValueError("ARFCN must be between 0 and 327916.")
+        self._channel = channel
```

Comment:
Because we accept Decimal, we could create an ARFCN that is not an integral value: `ARFCN(Decimal("3.5"))`.